fdbserver/Ratekeeper.actor.cpp

```diff
@@ -158,17 +158,28 @@ class RatekeeperImpl {
 	ACTOR static Future<Void> trackStorageServerQueueInfo(ActorWeakSelfRef<Ratekeeper> self,
 	                                                      StorageServerInterface ssi) {
 		self->storageQueueInfo.insert(mapPair(ssi.id(), StorageQueueInfo(self->id, ssi.id(), ssi.locality)));
-		TraceEvent("RkTracking", self->id)
+		self->healthMetrics.storageStats[ssi.id()] = HealthMetrics::StorageStats();
+		TraceEvent("RkTrackStorageStart", self->id)
 		    .detail("StorageServer", ssi.id())
 		    .detail("Locality", ssi.locality.toString());
-		try {
-			loop {
+
+		loop {
+			try {
 				ErrorOr<StorageQueuingMetricsReply> reply = wait(ssi.getQueuingMetrics.getReplyUnlessFailedFor(
 				    StorageQueuingMetricsRequest(), 0, 0)); // SOMEDAY: or tryGetReply?
 				Map<UID, StorageQueueInfo>::iterator myQueueInfo = self->storageQueueInfo.find(ssi.id());
+				ASSERT(myQueueInfo != self->storageQueueInfo.end());
 				if (reply.present()) {
 					myQueueInfo->value.update(reply.get(), self->smoothTotalDurableBytes);
 					myQueueInfo->value.acceptingRequests = ssi.isAcceptingRequests();
+
+					// Update health stats.
```

Comment:
A quick question: why do we want to update these health stats here? Seems that the those are updated in `void Ratekeeper::updateRate(RatekeeperLimits* limits)`

Reply:
The health metrics need to be updated when there's a new response. It seems to me that the code is not well structured as the metrics is not used anywhere in `Ratekeeper::updateRate(RatekeeperLimits* limits)`